app/src/main/java/com/akp/ceg4110/quickreports/DatabaseAccessor.java

```diff
@@ -0,0 +1,234 @@
+    public static final String NAME_COLUMN = "name";
+    public static final String DESCRIPTION_COLUMN = "description";
+    public static final String INCIDENT_WEATHER = "weather";
+
+    public static final String PICTURE_TABLE  = "image_table";
+    public static final String PICTURE_NAME_COLUMN = "name";
+    public static final String PICTURE_PRIMARY_COLUMN = "picture_reference";
+    public static final String PICTURE_COLUMN = "picture";
+
+    private SQLiteDatabase db;
+
+
+    public DatabaseAccessor(SQLiteDatabase db){
+        this.db = db;
+        //Build string for creating the incident_table table
+        //TEMPLATE:
+        //CREATE TABLE IF NOT EXISTS incident_table (name VARCHAR(255), description VARCHAR(255), PRIMARY KEY (name));
+        String createIncidentTable = String.format("CREATE TABLE IF NOT EXISTS %1$s (%2$s VARCHAR(255), %3$s VARCHAR(255), %4$s VARCHAR(500), PRIMARY KEY(%5$s));",
+                INCIDENT_TABLE, NAME_COLUMN, DESCRIPTION_COLUMN, INCIDENT_WEATHER, NAME_COLUMN);
+        //Build string for creating the image_table table
+        //TEMPLATE:
+        //CREATE TABLE IF NOT EXISTS image_table (picture_reference INTEGER PRIMARY KEY AUTOINCREMENT, name VARCHAR(255), picture BLOB);
+        String createPictureTable = String.format("CREATE TABLE IF NOT EXISTS %1$s ( %2$s INTEGER PRIMARY KEY AUTOINCREMENT, %3$s VARCHAR(255), %4$s BLOB);",
+                PICTURE_TABLE, PICTURE_PRIMARY_COLUMN, PICTURE_NAME_COLUMN, PICTURE_COLUMN, PICTURE_PRIMARY_COLUMN);
+        try{
+            db.execSQL(createIncidentTable); //Add incident table to DB
+            db.execSQL(createPictureTable);   //Add picture table to DB
+        }catch(Exception e){
+            System.out.println("Error creating table: " + e.getMessage());
+        }
+    }
+
+    public void addIncident(Incident incident){
+        //TEMPLATE:
+        //INSERT INTO incident_table VALUES ('incident.getName()', 'incident.getDescription()');
+        //Also written as:
+        //INSERT INTO incident_table VALUES (incident.toString());
+        //Doesn't include the images
+        String insertIncidentTable = String.format("INSERT INTO %1$s VALUES ('%2$s', '%3$s', '%4$s');",
+                INCIDENT_TABLE, incident.getName(), incident.getDescription(), incident.getWeather());
+        List<Bitmap> images = incident.getImages();
+        try{
+            db.execSQL(insertIncidentTable);
+            //Add pictures to picture table
+            for(int i = 0; i < images.size(); i++) {
+                ContentValues imageNameInsert = new ContentValues();
+                imageNameInsert.put(PICTURE_NAME_COLUMN, incident.getName());
+                imageNameInsert.put(PICTURE_COLUMN, imageToByte(images.get(i)));
+                db.insert(PICTURE_TABLE, null, imageNameInsert);
+            }
+        }catch(Exception e){
+            System.out.println("Error inserting incident into DB: " + e.getMessage());
```

Comment:
Just an FYI, console printouts won't do anything in an Android application.

If we want to display something to the user, we would have to display a toast, snackbar, or dialog.

I would think this should be handled by the calling item - so rather than catching the error it should be thrown I believe, so maybe add a throws clause in the method signature or just don't catch and let it throw the exception.

Reply:
I have the printlns because I was new to this when I was testing things out, I can change them later.